test.ts

```diff
@@ -0,0 +1,139 @@
+import { which } from "./deps/deno_which.ts"
+import { unimplemented } from "./deps/std/assert.ts"
+import { parse as parseFlags } from "./deps/std/flags.ts"
+import { blue, dim, gray, green, red, yellow } from "./deps/std/fmt/colors.ts"
+import { walk } from "./deps/std/fs.ts"
+import { Buffer, readLines } from "./deps/std/io.ts"
+import * as path from "./deps/std/path.ts"
+import { readerFromStreamReader, writeAll } from "./deps/std/streams.ts"
+import { parse as parseFrontmatter } from "./frontmatter.ts"
+
+const { _, reload, ...rest } = parseFlags(Deno.args, {
+  alias: {
+    c: "concurrency",
+    p: "project",
+    r: "reload",
+  },
+  string: ["concurrency", "project", "reload"],
+  default: {
+    concurrency: Infinity,
+  },
+})
+
+const include: string[] = []
+for await (
+  const { path: pathname } of walk(".", {
+    exts: [".ts", ".tsx"],
+    followSymlinks: true,
+    includeDirs: false,
+    match: _.map((value) => {
+      if (typeof value !== "string") {
+        throw new Error(
+          `Specified an invalid include \`${pathname}\` (expected a glob or path to example file)`,
```

Comment:
this doesn't work because `pathName` isn't actually defined here as it is only defined after this function is invoked.

```
error: Uncaught ReferenceError: Cannot access 'pathname' before initialization
          `Specified an invalid include \`${pathname}\` (expected a glob or path to example file)`,
                                            ^
    at https://raw.githubusercontent.com/paritytech/trun/reworking/test.ts:32:45
    at Array.map (<anonymous>)
    at https://raw.githubusercontent.com/paritytech/trun/reworking/test.ts:29:14
vscode ➜ /workspaces/capi (main) $ 
```